docs/api/json.md

```diff
@@ -0,0 +1,291 @@
+- via [`/ingress`](/api/push-data-to-a-sql-table) and
+  [`/egress`](/api/subscribe-to-a-stream-of-updates-from-a-sql-view-or-table)
+  REST endpoints, or
+- as a payload received from or sent to a connector
+
+Here we document the JSON format supported by Feldera.  The specification
+consists of four parts:
+
+1. [Encoding invividual table rows](#encoding-individual-rows). Describes
+   JSON encoding of an individual row in a SQL table or view.
+2. [Encoding data change events](#encoding-data-change-events).
+   A data change event represents an insertion, deletion, or modification
+   of a single row.
+3. [Encoding multiple changes](#encoding-multiple-changes). Describes JSON
+   encoding of a stream of data change events.
+4. [Configuring JSON event streams](#configuring-json-event-streams).
+   Describes how the user can specify the JSON format for a stream of events
+   when sending and receiving data via connectors or over HTTP.
+
+## Encoding individual rows
+
+A row in a SQL table or view is encoded as a JSON object whose keys match
+the names of the columns in the table.  Keys can occur in an arbitrary order.
+Column names are case-insensitive, except for columns whose names are declared
+in double quotes (e.g., `create table foo("col")`).  `NULL`s are encoded
+as JSON `null` values or by simply omitting the columns whose value is `NULL`.
+
+For example, given the following table declaration
+
+```sql
+create table json_test (
+    b BOOLEAN,
+    i INTEGER,
+    d DOUBLE,
+    v VARCHAR(32),
+    cc CHAR(16),
+    t TIME,
+    ts TIMESTAMP,
+    dt DATE,
+    ar BIGINT ARRAY
+);
+```
+
+the JSON encoding of a row would look like this:
+
+```json
+{
+    "B":true,
+    "I":-1625240816,
+    "D":0.7879946935782574,
+    "V":"quod",
+    "CC":"voluptatem",
+    "T":"05:05:24",
+    "TS":"2023-11-21 23:19:09",
+    "DT":"2495-03-07",
+    "AR":[1,2,3,4,5]
+}
+```
+
+## Types
+
+| Type                                    | Example                                         |
+|-----------------------------------------|-------------------------------------------------|
+| BOOLEAN                                 | `true`, `false`                                 |
+| TINYINT,SMALLINT, INTEGER, BIGINT       |  `1`, `-9`                                      |
+| FLOAT, DOUBLE, DECIMAL                  | `-1.40`, `12.53`, `1e20`                        |
+| VARCHAR, CHAR, STRING                   | `abc`                                           |
+| TIME                                    | `12:12:33`, `23:59:29.483`, `23:59:09.483221092`|
+| TIMESTAMP                               | `2024-02-25 12:12:33`                           |
+| DATE                                    | `2024-02-25`                                    |
+| BIGINT ARRAY                            | `[1, 2]`                                        |
+| VARCHAR ARRAY ARRAY                     | `[[ 'abc', '123'], ['c', 'sql']]`               |
+
+### `BOOLEAN`
```

Comment:
seems like the next few sections are duplication of the CSV format spec? maybe we can include it somehow to avoid duplication?

Reply:
This works, but there's a bunch of differences, e.g., in strings, arrays, and floats, so I think I'll just keep them separate for now.